spec/src/main/asciidoc/ch03-entity-operations.adoc

```diff
@@ -2802,26 +2797,34 @@ The possible flush modes are enumerated by `FlushModeType`.
 include::../../../../api/src/main/java/jakarta/persistence/FlushModeType.java[lines=18..-1]
 ----
 
-When a query is executed within a transaction:
-
-- If `FlushModeType.AUTO` is set via the `Query`, `TypedQuery`, or
-  `StoredProcedureQuery` object, or if the flush mode setting for
-  the persistence context is `AUTO` (the default) and a flush
-  mode setting has not been specified for the query object, the
-  persistence provider must ensure that every modification to the
-  state of every entity associated with the persistence context
-  which could possibly affect the result of the query is visible
-  to the processing of the query. The persistence provider
-  implementation might guarantee this by flushing pending updates
-  to modified entities to the database before executing the query.
-
-- On the other hand, if `FlushModeType.COMMIT` is set, the effect
-  of updates made to entities in the persistence context on query
-  results is unspecified.
+When a Jakarta Persistence or native SQL query is executed within
+a transaction via a `Query`, `TypedQuery`, or `StoredProcedureQuery`
+object obtained from an `EntityManager` joined to the transaction,
+the effective flush mode is determined by the current flush mode of
+the `Query` object, if any flush mode has been specified by calling
+`setFlushMode()`, or, otherwise, by the current flush mode of the
+persistence context.
 
-If there is no transaction active or if the persistence context has
-not been joined to the current transaction, the persistence provider
-must not flush to the database, regardless of the flush mode.
+- If `FlushModeType.AUTO` is in effect, the persistence provider
+  must ensure that every modification to the state of every entity
+  associated with the persistence context which could possibly
+  affect the result of the query is visible to the processing of
+  the query. The persistence provider implementation might
+  guarantee this by flushing pending updates to modified entities
+  to the database before executing the query.
+
+- Otherwise, if `FlushModeType.COMMIT` is set, the effect on the
+  results of the query of pending modifications to entities in
+  the persistence context is unspecified.
```

Comment:
Rearranged wording for minor improvement in readability,
```suggestion
- Otherwise, if `FlushModeType.COMMIT` is set, it is unspecified
  how pending modifications to entities in the persistence context
  effect the results of the query.
```

Reply:
affect not effect

```suggestion
- Otherwise, if `FlushModeType.COMMIT` is set, it is unspecified
  how pending modifications to entities in the persistence context
  affect the results of the query.
```